template/python-afterburn/index.py

```diff
@@ -0,0 +1,106 @@
+"""python template for running in afterburn mode"""
+
+import sys
+from function import handler
+
+SEPARATOR = "\r\n"
+
+def parse_header():
+    """Reads the http headers from watchdog over stdin
+    Returns:
+        string: http method of the request
+        dict: http headers in string to string dictionary
+    """
+    raw_header = ""
+    while True:
+        line = read_line()
+        if line == "\n":
+            break
+        raw_header += line
+
+    headers = {}
+    method = ''
+    for index, header in enumerate(raw_header.split("\n")):
+        # Parse the first line to get http method
+        if index == 0:
+            method = header.split(' ')[0]
+            continue
+        # Parse rest of the lines for key, value
+        parts = header.split(':')
+        headers[parts[0]] = ''.join(parts[1:])
+
+    return method, headers
+
+def read_line():
+    """Reads the individual line representing header from stdin"""
+    line = ""
+    while True:
+        char = sys.stdin.read(1)
+        if char == "\r":
+            continue
+        line += char
+        if char == "\n":
+            break
+    return line
+
+def get_request():
+    """Reads the http request from watchdog over stdin
+
+    Returns:
+        string: http method of the request
+        dict: http headers in string to string dictionary
+        string: http body of the request
+    """
+    method, headers = parse_header()
+
+    if 'Content-Length' not in headers:
+        raise KeyError('Content-Length must be present in the request')
+
+    content_length = int(headers['Content-Length'])
+    body = sys.stdin.read(content_length)
```

Comment:
What should we do if the HTTP request is deliberately malformed? We could do this for instance with `netcat` - i.e. the body is empty or too short. What will happen?

Reply:
@alexellis @viveksyngh 

We can add a custom RequestMalformedError and use it to throw any error related to request. 

Upon catching it we can return 500 with body like `"Malformed Request: {}".format(' - '.join(error.args))`